src/io/flutter/settings/FlutterSettings.java

```diff
@@ -217,6 +227,25 @@ public void setOpenInspectorOnAppLaunch(boolean value) {
     fireEvent();
   }
 
+  public boolean useBazelByDefault() {
+    return Registry.is(ignorePubspecRegistryKey, false);
+  }
+
+  public void setUseBazelByDefault(boolean value) {
+    Registry.get(ignorePubspecRegistryKey).setValue(value);
+
+    fireEvent();
+  }
+
+  /**
+   * Determines whether bazel should be used for a project based on the user's preferences and the structure of the project.
+   */
+  public boolean shouldUseBazel(Project project) {
+    final boolean isBazelProject = FlutterModuleUtils.isFlutterBazelProject(project);
+    final boolean isNotPubProject = PubRoots.forProject(project).isEmpty();
+    return isBazelProject && (useBazelByDefault() || isNotPubProject);
```

Comment:
Why not just `return Registry.is(ignorePubspecRegistryKey, false);`  Is there some usecase where this complexity will help?

Reply:
Yes, this is useful:

 - If we ask for the value of the setting in a pub-only project, then we should not attempt to use bazel.
 - Conversely, if this is not set in a bazel-only project, then we should still always use bazel.